proposals/NNNN-experiment-tracking/README.md

```diff
@@ -0,0 +1,507 @@
+   could map to a trial. Each run has a name, status, timestamps, source (e.g. pipelines), the user that created it, and
+   can have associated artifacts, metrics, parameters, and arbitrary metadata/tags as key-value pairs.
+1. **Nested runs**: When a run represents a workflow with multiple sequential and/or parallel steps (such as a
+   pipeline), a nested run represents each individual step. These are essentially runs with a parent. In pipelines, each
+   DAG node is represented by one or more nested runs. A nested run's parent can also be another nested run.
+1. **Metrics**: A key-value pair where the key is the metric name and the value is a numeric value.
+1. **Parameters**: A key-value pair where the key is the parameter name. This maps to a run ID.
+1. **Generic artifacts**: An artifact with a name, URI, and metadata. This would be used to represent files such as JSON
+   files, logs, CSV files, and other data files.
+1. **Unregistered Models**: Model Registry currently has the concept of registered model versions, but there is an
+   intermediate state that needs to be represented where a run produces an output model that isn't yet registered. It
+   has a name, URI, and arbitrary metadata/tags as key-value pairs.
+1. **Datasets**: An artifact used for representing input data for training, evaluation, RAG, and other machine learning
+   tasks.
+1. **Images**: An artifact used to represent an image to be rendered directly in the browser. This will be mainly used
+   for custom visualizations not available natively in the UI.
+
+### Filtering Requirements
+
+Some basic level of filtering is required for initial UI rendering. This includes but is not limited to:
+
+- Filter all entities by namespace
+- Filter runs by experiment name
+- Filter runs by parameter name
+- Filter runs by metric name
+- Filter runs by parent run
+- Filter runs by status
+- Filter runs by metadata fields/tags
+- Filter runs by source
+- Filter all entities by user that created it
+- Filter by timestamps (before, after)
+- Fuzzy searching on entity names
+
+### Multi-tenancy Implementation
+
+Model Registry's current authorization model gates access at the API level through a proxy rather than at the individual
+resource or namespace level. This approach prevents the same Model Registry instance from being shared across teams that
+require isolation. This limitation conflicts with Kubeflow Pipelines' multiuser mode, which is the default deployment
+strategy for the Kubeflow platform. MLMD has the same limitation, so adding multi-tenancy support to Model Registry
+would address this gap for all of Kubeflow's metadata needs.
+
+This proposal adds an authorization mode to Model Registry that leverages Kubernetes subject access review. In this
+mode, all entities would be mapped to Kubernetes namespaces on the cluster to provide namespace-level isolation. For
```

Comment:
We cannot go back to the namespace scoped limitation of mlmd architecture before. A registered model is not tied to a specific namespace, but shared across all clients within a higher level tenant scope. Model Registry will implement multi-tenancy but use Kubernets user groups and roles to do so, which decouples client location (namespace) from resource location.

Reply:
The proposal is to have an authorization mode that is namespaced. So in that mode, a namespace would be required on all entities. When not in that mode, it's not a field that is exposed/accepted by the API.

We could start out by having this mode only apply to the new experiment tracking entities but still have the existing entities be global to the instance.